chrome/content/zotero/xpcom/zotero.js

```diff
@@ -1095,6 +1095,13 @@ Services.scriptloader.loadSubScript("resource://zotero/polyfill.js");
 	 */
 	this.launchFile = function (file) {
 		file = Zotero.File.pathToFile(file);
+
+		var customPDFReaderPath  = Zotero.Prefs.get("extensions.zotero.pdf.launcher");
+		if(file.path.toUpperCase().endsWith(".PDF") && customPDFReaderPath){
+            launchFileWithGivenApplication(customPDFReaderPath, file);
```

Comment:
Pass `file.path` instead of `file` — we're trying to avoid `nsIFile` in new code as much as possible.